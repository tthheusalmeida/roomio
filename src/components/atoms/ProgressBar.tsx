interface ProgressBarProps {
  value: number; // de 0 a 100
  height?: string;
  color?: string;
  showLabel?: boolean;
}

export default function ProgressBar({
  value,
  height = "h-2",
  color = "bg-blue-500",
  showLabel = false,
}: ProgressBarProps) {
  return (
    <div className={`w-full bg-gray-200 rounded ${height} overflow-hidden`}>
      <div
        className={`transition-all duration-300 ${color} ${height}`}
        style={{ width: `${value}%` }}
      />
      {showLabel && (
        <p className="text-sm text-center text-gray-500 mt-1">
          {Math.round(value)}%
        </p>
      )}
    </div>
  );
}
