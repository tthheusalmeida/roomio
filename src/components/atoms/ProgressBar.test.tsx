import { render, screen } from "@testing-library/react";
import ProgressBar from "./ProgressBar";

describe("ProgressBar component", () => {
  it("renders the progress bar with correct width", () => {
    const { container, asFragment } = render(<ProgressBar value={50} />);

    const innerBar = container.firstChild?.firstChild as HTMLElement; // a div interna
    expect(innerBar).toHaveStyle({ width: "50%" });

    expect(asFragment()).toMatchSnapshot();
  });

  it("applies custom height and color", () => {
    const { container, asFragment } = render(
      <ProgressBar value={75} height="h-4" color="bg-red-500" />
    );

    const innerBar = container.firstChild?.firstChild as HTMLElement;
    expect(innerBar).toHaveClass("h-4", "bg-red-500");
    expect(innerBar).toHaveStyle({ width: "75%" });

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders label when showLabel is true", () => {
    render(<ProgressBar value={33} showLabel />);
    expect(screen.getByText("33%")).toBeInTheDocument();
  });

  it("does not render label when showLabel is false", () => {
    render(<ProgressBar value={33} showLabel={false} />);
    expect(screen.queryByText("33%")).not.toBeInTheDocument();
  });
});
