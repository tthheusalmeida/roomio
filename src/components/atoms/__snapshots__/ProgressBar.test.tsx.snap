// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ProgressBar component applies custom height and color 1`] = `
<DocumentFragment>
  <div
    class="w-full bg-gray-200 rounded h-4 overflow-hidden"
  >
    <div
      class="transition-all duration-300 bg-red-500 h-4"
      style="width: 75%;"
    />
  </div>
</DocumentFragment>
`;

exports[`ProgressBar component renders the progress bar with correct width 1`] = `
<DocumentFragment>
  <div
    class="w-full bg-gray-200 rounded h-2 overflow-hidden"
  >
    <div
      class="transition-all duration-300 bg-blue-500 h-2"
      style="width: 50%;"
    />
  </div>
</DocumentFragment>
`;
